### ISSUE TITLE
Revert deletion of divide by zero error 

### ISSUE THREAD

**Issue by xylosper** [google/closure-compiler#2001]
I asked #1997 issue and got the answer which explained that divide-by-zero check was deleted by 25829b0395164533782d608399096803321225a7. But I cannot agree with the commit message:

>I've never heard of this catching any real bug. It seems like if
you explicitly typed "1 / 0", you meant to do that.

As far as I know, closure compiler can handle some static operations, for instance, closure compiler produces JSC_INDEX_OUT_OF_BOUNDS_ERROR for next source code:

```js
let x = 0;
let y = 4;
let array = [1, 2, 3];
alert(array[x + y]);
```

Static divide-by-zero error occurs in the same manner. No one would write such a code `1 / 0` because people would use `NaN` for not a number and `Infinity` for infinite number.

Instead, I expect that closure compiler checks for me such code:

```js
let x = 1;
...
let y = 1 / (x >> 1); // maybe it would be corrected to (x << 1) or ((x >> 1) + 1)
...
```

And, in fact, I've done such try and error before which put me to shame.

I think this feature is totally valid and useful, so please revert the commit 25829b0395164533782d608399096803321225a7.

**Comment by dimvar** [comment]
It's true that every once in a while this may find an issue, but that should happen very rarely, which is why the code was deleted. To avoid bloating our code base, we have to pick and choose what we warn about.
